CRM/Contribute/Form/Contribution.php

```diff
@@ -1021,6 +1046,23 @@ public static function formRule($fields, $files, $self) {
       $errors['financial_type_id'] = ' ';
       $errors['_qf_default'] = $e->getMessage();
     }
+    if (!$self->_mode && !empty($fields['pan_truncation'])
+    ) {
+      $contributionDoneViaPaymentProcesor = FALSE;
+      if ($self->_id) {
+        $financialTrxn = $self->getlatestPayments();
+        if (!empty($financialTrxn['financial_trxn_id.payment_processor_id'])) {
+          $contributionDoneViaPaymentProcesor = TRUE;
+        }
+      }
+      if (!$contributionDoneViaPaymentProcesor
```

Comment:
Typo on this variable name - please change to $contributionDoneViaPaymentProcessor (double S in Processor)